samcli/lib/config/samconfig.py

```diff
@@ -39,11 +40,21 @@ def __init__(self, config_dir, filename=None):
             Optional. Name of the configuration file. It is recommended to stick with default so in the future we
             could automatically support auto-resolving multiple config files within same directory.
         """
+        self.document = {}
         self.filepath = Path(config_dir, filename or DEFAULT_CONFIG_FILE_NAME)
+        self.file_manager = self.FILE_MANAGER_MAPPER.get(self.filepath.suffix[1:], None)
```

Comment:
is there no native file extension api?

Reply:
if not, we can move the the filemanager mapper, to just have .toml, .yaml and so-on as the keys. having to do string splicing, makes me queasy on how things can break.

Reply:
The closest thing we can get with `pathlib.Path` is suffix, which returns the extension in the format of ".toml", with the period. In this file, extensions are usually referred to without the period, such as the case with `DEFAULT_CONFIG_FILE_EXTENSION`. Hence, anywhere I map extensions, such as the `FILE_MANAGER_MAPPER`, I use the extension name without the period.